lib/tasks/ingestor.rake

```diff
@@ -124,38 +124,50 @@ namespace :fromthepage do
     # first process all sub-directories
     ls = Dir.glob(File.join(temp_dir, "*"))
     ls.each do |path|
-      print "ingest_tree handling #{path})\n"
+      print "ingest_tree considering #{path})\n"
       if Dir.exist? path
         print "Found directory #{path}\n"
         ingest_tree(document_upload, path) #recurse
       end
     end    
     # now process this directory if it contains image files
-    image_files = Dir.glob(File.join(temp_dir, "*{"+IMAGE_FILE_EXTENSIONS.join(',')+"}"))
+    image_files = Dir.glob(File.join(temp_dir, "*.{"+IMAGE_FILE_EXTENSIONS.join(',')+"}"))
     if image_files.length > 0
-      print "Found image files in #{temp_dir} -- converting to a work\n"
+      print "Found #{image_files.length} image files in #{temp_dir} -- converting to a work\n"
       convert_to_work(document_upload, temp_dir)
+      print "Finished converting files in #{temp_dir} to a work\n"
     end
+    print "Finished ingest_tree for #{temp_dir}\n"
     
   end
   
   def convert_to_work(document_upload, path)
+    print "convert_to_work creating database record for #{path}\n"
+    print "\tconvert_to_work owner = #{document_upload.user.login}\n"
+    print "\tconvert_to_work collection = #{document_upload.collection.title}\n"
+    print "\tconvert_to_work title = #{File.basename(path).ljust(3,'.')}\n"
+#    binding.pry if path == "/tmp/fromthepage_uploads/16/terrell-papers-jpg"
     User.current_user=document_upload.user
     
     work = Work.new
     work.owner = document_upload.user
     work.collection = document_upload.collection
-    work.title = File.basename(path)
+    work.title = File.basename(path).ljust(3,'.')
```

Comment:
I believe that this was the source of the "processing" bug.  At the very end of the collection ingestion, the depth-first-search considers the root directory.  Because of the ends-with-jpg bug, it was trying to ingest the root temp directory, `/tmp/fromthepage_uploads/16`.  It turns out that we require works to have titles of at least three characters, so validation was failing.

Adding `ljust` pads out a work title with periods until it's at least three characters.  That fixed the `processing` bug, since `ingest_tree` actually completes.  Now that we've fixed the ends-in-jpg bug as well, this should not be needed, but I didn't see any reason to remove it.